### PR TITLE
Fix #3459: Merge all topics into single XML file

### LINKED ISSUE #3459: Merge all_topics.xml into single xml file
Currently there are 4 versions of `all_topics.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

### PR DESCRIPTION
**Pull request by deepanshu731**
Fix  #3459 
I have merged all_topics.xml  into single file. I have added values in dimens.xml according to screen sizes.

**Screenshots before merging into single file.**

Screenshot from 2021-07-24 19-02-36

Screenshot from 2021-07-24 19-02-59

Screenshot from 2021-07-24 19-05-15

Screenshot from 2021-07-24 19-05-49

**Screenshots after merging into single file**

Screenshot from 2021-07-24 18-53-15

Screenshot from 2021-07-24 18-54-07

Screenshot from 2021-07-24 18-55-19

Screenshot from 2021-07-24 18-55-47

### CODE REVIEW

**Comment by deepanshu731** [comment]
I have already filled the CLA but its still showing error. plz help @rt4914

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="false" project-jdk-name="1.8" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
   </component>
   <component name="ProjectType">
     <option name="id" value="Android" />
   </component>
```
Comment (prayutsu) [inline .idea/misc.xml]: @deepanshu731 These are unwanted, please revert these changes.

File: .idea/runConfigurations.xml [hunk 2]
```diff
@@ -3,10 +3,11 @@
...
       <set>
+        <option value="com.android.tools.idea.compose.preview.runconfiguration.ComposePreviewRunConfigurationProducer" />
         <option value="org.jetbrains.plugins.gradle.execution.test.runner.AllInPackageGradleConfigurationProducer" />
         <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestClassGradleConfigurationProducer" />
         <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestMethodGradleConfigurationProducer" />
       </set>
     </option>
   </component>
-</project>
+</project>
```
Comment (prayutsu) [inline .idea/runConfigurations.xml]: ditto

File: build.gradle [hunk 3]
```diff
@@ -8,7 +8,7 @@ buildscript {
     gradlePluginPortal()
   }
   dependencies {
-    classpath 'com.android.tools.build:gradle:3.6.1'
+    classpath 'com.android.tools.build:gradle:4.2.2'
     classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
```
Comment (prayutsu) [inline build.gradle]: Ditto here.

**Review by prayutsu — CHANGES_REQUESTED** [review]
@deepanshu731 Left some comments. PTAL.

**Comment by prayutsu** [comment]
@deepanshu731 Also, please link the issue to the PR as mentioned in the PR checklist.

File: .idea/misc.xml (hunk 1 above)
Comment (deepanshu731) [reply .idea/misc.xml]: done

File: .idea/runConfigurations.xml (hunk 2 above)
Comment (deepanshu731) [reply .idea/runConfigurations.xml]: done

File: build.gradle (hunk 3 above)
Comment (deepanshu731) [reply build.gradle]: done

**Comment by deepanshu731** [comment]
@prayutsu done required changes , review it again

**Comment by deepanshu731** [comment]
> @deepanshu731 Also, please link the issue to the PR as mentioned in the PR checklist.

done

**Review by prayutsu — APPROVED** [review]
LGTM, thanks @deepanshu731.
Also, note that the conversation thread must only be resolved by the author of the conversation e.g I left some comments, so they were meant to be resolved only by me.

File: app/src/main/res/values/dimens.xml [hunk 4]
```diff
@@ -434,4 +434,11 @@
   <dimen name="concept_card_explanation_text_marginTop">28dp</dimen>
   <dimen name="concept_card_explanation_text_marginEnd">40dp</dimen>
 
+  <!-- All Topics -->
+  <dimen name="all_topics_text_marginStart">28dp</dimen>
+  <dimen name="all_topics_text_marginTop">56dp</dimen>
+  <dimen name="all_topics_text_marginEnd">32dp</dimen>
+  <dimen name="all_topics_text_marginBottom">12dp</dimen>
+
+
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml:442]: Remove this extra blank line

File: app/src/main/res/values-land/dimens.xml [hunk 5]
```diff
@@ -253,4 +253,10 @@
   <dimen name="concept_card_explanation_text_marginTop">24dp</dimen>
   <dimen name="concept_card_explanation_text_marginEnd">64dp</dimen>
 
+  <!-- All Topics -->
+  <dimen name="all_topics_text_marginStart">72dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: rename this to `all_topics_text_margin_start`

Same applies for all values that you have introduced.

I understand that the convention that you used is part of `concept_card` but its incorrect. We should file an issue for that. Can you file an issue and mention it here?

**Review by rt4914 — CHANGES_REQUESTED** [review]
@deepanshu731 PTAL

File: app/src/main/res/values/dimens.xml (hunk 4 above)
Comment (deepanshu731) [reply app/src/main/res/values/dimens.xml:442]: done

File: app/src/main/res/values-land/dimens.xml (hunk 5 above)
Comment (deepanshu731) [reply app/src/main/res/values-land/dimens.xml]: done changes and also created the issue #3530

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.
